pkgs/misc/wine-nine-standalone/setup_nine.sh

```diff
@@ -0,0 +1,130 @@
+#!@bash@/bin/bash -e
+
+# Loosely based on dxvk_setup.sh
+
+set -eu -o pipefail
+
+nine32=@nine32@
+nine64=@nine64@
+
+usage() {
+    echo "wine-nine-standalone @version@"
+    echo "Set up Wine prefix with Nine DLLs"
+    echo
+    echo "USAGE"
+    echo "    $0 [COMMAND] [OPTIONS]"
+    echo
+    echo "COMMANDS"
+    echo "    install                  Link the Nine DLLs into the prefix"
+    echo "    uninstall                Remove Nine DLLs from the prefix"
+    echo "    enable (default)         Replace d3d9.dll with Nine's"
+    echo "    disable                  Restore the stock Wine DLLs in the prefix"
+    echo
+    echo "OPTIONS"
+    echo "    -h, --help               Show this help message"
+    exit 1
+}
+
+action=enable
+while [[ -n "${1:-}" ]]; do
+    case "$1" in
+        install|uninstall|enable|disable)
+            action=$1
+            ;;
+        -h|--help)
+            usage
+            ;;
+        *)
+            echo "Unrecognized option: $1"
+            usage
+            ;;
+    esac
+    shift
+done
+
+export WINEPREFIX=${WINEPREFIX:-"$HOME/.wine"}
+
+# Check wine prefix before invoking wine, so that we don't accidentally create one
+if [[ ! -f "$WINEPREFIX/system.reg" ]]; then
+    echo "$WINEPREFIX: not a valid wine prefix." >&2
+    exit 1
+fi
+
+export WINEDEBUG=-all
+# disable mscoree and mshtml to avoid downloading
+# wine gecko and mono
+export WINEDLLOVERRIDES="mscoree,mshtml="
+
+# Ensure we're only calling binaries from the same Wine install
+wine_path=$(dirname -- "$(command -v wineboot)")
+
+update_prefix=false
+
+common_setup() {
+    sys_path=$($wine winepath -u 'C:\windows\system32' 2> /dev/null)
+    sys_path="${sys_path/$'\r'/}"
+}
+
+link() {
+    ln -sf "$1" "$2"
+    echo "Linked $1 to $2"
+}
+
+remove() {
+    echo "Removing $1"
+    rm -f "$1"
+}
+
+install() {
+    link "$nine"/lib/d3d9-nine.dll.so "$sys_path"/d3d9-nine.dll
+    # Check that Nine is actually working
+    if ! "$wine" "$nine"/bin/ninewinecfg.exe.so -p; then
+        "$0" uninstall
+        exit 1
+    fi
+}
+
+uninstall() {
+    if readlink -- "$sys_path"/d3d9.dll > /dev/null; then
+        disable
+    fi
+    remove "$sys_path"/d3d9-nine.dll
+}
+
+enable() {
+    install
+    link d3d9-nine.dll "$sys_path"/d3d9.dll
+}
+
+disable() {
+    remove "$sys_path"/d3d9.dll
+    update_prefix=true
+}
+
+found_wine=false
+
+wine=$wine_path/wine
+if [[ -x "$wine" ]]; then
+    found_wine=true
+    nine=$nine32
+    common_setup
+    $action
+fi
+
+wine=$wine_path/wine64
+if [[ -x "$wine" ]]; then
+    found_wine=true
+    nine=$nine64
+    common_setup
+    $action
+fi
```

Comment:
One small problem I can see here is that in the 32 bit version of the wrapper, `$nine64` is actually going to point to the 32 bit version. This means that if you somehow end up with the 32 bit wrapper and wineWoW in your environment, you would install 32 bit Nine to the 64 bit path in the prefix. Pure 64 bit builds of Wine would probably fail to run ninewinecfg so it's not really a problem there.

Do we care?